gymnasium/spaces/text.py

```diff
@@ -4,12 +4,12 @@
 from typing import Any
 
 import numpy as np
-import numpy.typing as npt
+from numpy.typing import NDArray
 
 from gymnasium.spaces.space import Space
 
 
-alphanumeric: frozenset[str] = frozenset(
+alphanumeric: set[str] = set(
```

Comment:
Why change from frozenset to set?

Reply:
Because `charset: set[str] | str = alphanumeric,` in args, so it was either accepting `frozenstr[str]` as type or doing the uppder thing. 

Since frozen set is already used after 
```
        self._char_set: frozenset[str] = frozenset(charset)
        self._char_list: tuple[str, ...] = tuple(charset)
        self._char_index: dict[str, np.int32] = {
            val: np.int32(i) for i, val in enumerate(tuple(charset))
        }
        self._char_str: str = "".join(sorted(tuple(charset)))
```
I didn't know which solutions to pick

Reply:
I think that frozensets are better than the sets due to their immutability in particular for class attributes that shouldn't change after being assigned